cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -362,32 +362,39 @@ func ValidateKubeconfigsForExternalCA(outDir string, cfg *kubeadmapi.InitConfigu
 }
 
 func getKubeConfigSpecsBase(cfg *kubeadmapi.InitConfiguration) (map[string]*kubeConfigSpec, error) {
-	apiServer, err := kubeadmutil.GetControlPlaneEndpoint(cfg.ControlPlaneEndpoint, &cfg.LocalAPIEndpoint)
+	controlPlaneEndpoint, err := kubeadmutil.GetControlPlaneEndpoint(cfg.ControlPlaneEndpoint, &cfg.LocalAPIEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	localAPIEndpointIP, localAPIEndpointPort, err := kubeadmutil.ParseAPIEndpoint(&cfg.LocalAPIEndpoint)
```

Comment:
What about implementing `kubeadmutil.GetLocalAPIEndpoint` so we are consistent with the same path used above?

Reply:
SGTM, i will add this extra utility function.

Reply:
updated.